projects/packages/stats-admin/src/class-notices.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * A class that handles the notices for the Stats Admin dashboard.
+ *
+ * @package automattic/jetpack-stats-admin
+ */
+
+namespace Automattic\Jetpack\Stats_Admin;
+
+use Automattic\Jetpack\Stats\Options as Stats_Options;
+
+/**
+ * The Notices class handles the notices for the Stats Admin dashboard.
+ *
+ * @package Automattic\Jetpack\Stats_Admin
+ */
+class Notices {
+	const OPT_OUT_NEW_STATS_NOTICE_ID  = 'opt_out_new_stats';
+	const NEW_STATS_FEEDBACK_NOTICE_ID = 'new_stats_feedback';
+	const OPT_IN_NEW_STATS_NOTICE_ID   = 'opt_in_new_stats';
+
+	const NOTICE_STATUS_DISMISSED = 'dismissed';
+	const NOTICE_STATUS_POSTPONED = 'postponed';
+
+	const VIEWS_TO_SHOW_FEEDBACK = 3;
+	const POSTPONE_FEEDBACK_DAYS = 30;
+
+	/**
+	 * Update the notice status.
+	 *
+	 * @param mixed $id ID of the notice.
+	 * @param mixed $status Status of the notice.
+	 * @param int   $postponed_for Postponed for how many seconds.
+	 * @return bool
+	 */
+	public function update_notice( $id, $status, $postponed_for = self::POSTPONE_FEEDBACK_DAYS * DAY_IN_SECONDS ) {
+		$notices        = Stats_Options::get_option( 'notices' );
+		$notices[ $id ] = array(
+			'status'       => $status,
+			'id'           => $id,
+			'dismissed_at' => time(),
+		);
+		// Set the next show time if the notice is postponed.
+		if ( $status === self::NOTICE_STATUS_POSTPONED && $postponed_for > 0 ) {
+			$notices[ $id ]['next_show_at'] = time() + $postponed_for;
+		}
+		return Stats_Options::set_option( 'notices', $notices );
+	}
+
+	/**
+	 * For now, we support only one notice at a time.
+	 *
+	 * @return array
+	 */
+	public function get_notices_to_show() {
+		$new_stats_enabled = Stats_Options::get_option( 'enable_odyssey_stats' );
+
+		// Show opt in notice shown on legacy stats if new stats is disabled.
+		if ( ! $new_stats_enabled && ! $this->is_notice_hidden( self::OPT_IN_NEW_STATS_NOTICE_ID ) ) {
+			return array( self::OPT_IN_NEW_STATS_NOTICE_ID => true );
+		}
+
+		// Views > 3 and not hidden, show the feedback notice.
+		if ( $this->get_new_stats_views() >= self::VIEWS_TO_SHOW_FEEDBACK && ! $this->is_notice_hidden( self::NEW_STATS_FEEDBACK_NOTICE_ID ) ) {
+			return array(
+				self::NEW_STATS_FEEDBACK_NOTICE_ID => true,
+			);
+		}
+
+		// If opt-out notice is not hidden, we show it.
+		if ( ! $this->is_notice_hidden( self::OPT_OUT_NEW_STATS_NOTICE_ID ) ) {
+			return array(
+				self::OPT_OUT_NEW_STATS_NOTICE_ID => true,
+			);
+		}
```

Comment:
Would it be better to always return an array like the following?

```
return array(
	self::OPT_IN_NEW_STATS_NOTICE_ID => $first_conditional_here,
	self::NEW_STATS_FEEDBACK_NOTICE_ID => $second_conditional_here,
	self::OPT_OUT_NEW_STATS_NOTICE_ID => $third_conditional_here,
);
```

Or would this spawn multiple notices at the same time?

Reply:
I thought it would assure only one notice is returned in that way. I refactored with you recommendation, it works well and looks tidier. Thanks. @jsnmoon